aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPlugin.java

```diff
@@ -79,8 +93,71 @@ public void enable() {
      * {@inheritDoc}
      */
     @Override
-    public void identifyUser(@NonNull String userId, @NonNull AnalyticsProfile profile) {
-        throw new UnsupportedOperationException("This operation has not been implemented yet.");
+    public void identifyUser(@NonNull String userId, @NonNull AnalyticsUserProfile profile) {
+        EndpointProfile endpointProfile = targetingClient.currentEndpoint();
+        // Assign userId to the endpoint.
+        EndpointProfileUser user = new EndpointProfileUser();
+        user.setUserId(userId);
+        endpointProfile.setUser(user);
+        // Add user-specific data to the endpoint
+        addUserProfileToEndpoint(endpointProfile, profile);
+        // update endpoint
+        targetingClient.updateEndpointProfile();
+    }
+
+    /**
+     * Add user specific data from {@link AnalyticsUserProfile} to the endpoint profile.
+     * @param endpointProfile endpoint profile.
+     * @param userProfile user specific data to be added to the endpoint.
+     */
+    private void addUserProfileToEndpoint(EndpointProfile endpointProfile,
+                                          AnalyticsUserProfile userProfile) {
+        endpointProfile.addAttribute("email", Arrays.asList(userProfile.getEmail()));
+        endpointProfile.addAttribute("name", Arrays.asList(userProfile.getName()));
+        endpointProfile.addAttribute("plan", Arrays.asList(userProfile.getPlan()));
+        // Add location
+        addLocation(endpointProfile.getLocation(), userProfile.getLocation());
+        // Add custom properties
+        addCustomProperties(endpointProfile, userProfile.getCustomProperties());
+    }
+
+    /**
+     * Add custom user properties to the endpoint profile.
+     * @param endpointProfile endpoint profile.
+     * @param customProperties custom user properties to be added to the endpoint profile.
+     */
+    private void addCustomProperties(EndpointProfile endpointProfile,
+                                     Properties customProperties) {
+        if (customProperties != null) {
+            for (Map.Entry<String, Property<?>> entry : customProperties.get().entrySet()) {
+                if (entry.getValue() instanceof StringProperty) {
+                    endpointProfile.addAttribute(entry.getKey(),
+                            Arrays.asList(((StringProperty) entry.getValue()).getValue()));
+                } else if (entry.getValue() instanceof DoubleProperty) {
+                    endpointProfile.addMetric(entry.getKey(),
+                            ((DoubleProperty) entry.getValue()).getValue());
+                } else {
+                    Amplify.Hub.publish(HubChannel.ANALYTICS, HubEvent.create("Analytics.identifyUser",
```

Comment:
This behavior seems reasonable, but I'm not sold on the details of the `HubEvent`.

1. Can you get the event name into a shared `enum`? The other categories currently enumerate the possible event names in a central location, and have a pattern for what the event names should look like. Analytics should do likewise.

2. Since something exceptional has occurred, the payload type of the event should probably be an `AnalyticsException`. "Refer to the documentation" isn't an adequate recovery suggestion. If you'll point a user at documentation, it should include a link URL. But, the issue here is probably more like they tried to supply a property type that isn't supported, and so they should try to use one of the supported property types, instead. The supported property types are (state them here.)

Reply:
Added enum and updated the event payload to be an exception.